components/openvsx-proxy/pkg/handler.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package pkg
+
+import (
+	"fmt"
+	"math/rand"
+	"net/http"
+	"net/http/httputil"
+	"strconv"
+	"time"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+func (o *OpenVSXProxy) Handler(p *httputil.ReverseProxy) func(http.ResponseWriter, *http.Request) {
+	return func(rw http.ResponseWriter, r *http.Request) {
+		start := time.Now()
+
+		var (
+			hitCacheRegular = false
+			hitCacheBackup  = false
+		)
+
+		reqid := ""
+		uuid, err := uuid.NewRandom()
+		if err != nil {
+			log.WithError(err).Warn("cannot generate a UUID")
+			reqid = fmt.Sprintf("req%d", rand.Intn(999999))
+		} else {
+			reqid = uuid.String()
+		}
+
+		logFields := logrus.Fields{
+			LOG_FIELD_FUNC:           "request_handler",
+			LOG_FIELD_REQUEST_ID:     reqid,
+			LOG_FIELD_REQUEST:        fmt.Sprintf("%s %s", r.Method, r.URL),
+			"request_content_length": strconv.FormatInt(r.ContentLength, 10),
```

Comment:
do we send these headers to upstream open-vsx? It seems they needed only in callback functions? Should not we use r.WithContext(context.WithValue()) and r.Context().Value() instead to store additional information?

Reply:
These fields are internal logging fields. But I guess you mean these two header fields:

https://github.com/gitpod-io/gitpod/blob/be187030fbf333dbf749111ef5badf883226e081/components/openvsx-proxy/pkg/run.go#L21-L22

Indeed. That could work that we set these as context values. I'll try this.

Reply:
correct